Extend list of compilation options (#1386)
fluent/fluent-bit-docs: installation/sources/build-and-install.md

## Build Options

> Fluent Bit provides certain options to CMake that can be enabled or disabled when configuring, please refer to the following tables under the _General Options_, _Development Options_, Input Plugins _and _Output Plugins_ sections.

Reviewer: Since you're editing this line, it'd be good to get some style changes in place, both for line length, and to hew closer to our updated general style recommendations for Fluent Bit docs.

```suggestion
Fluent Bit provides options to CMake that you can enable or disable when configuring.
Refer to the following tables in the **General Options**, **Development Options**,
**Input Plugins** and **Output Plugins** sections.
```

| FLB_AWS | Enable AWS support | Yes |
| FLB_AWS_ERROR_REPORTER | Build with aws error reporting support | No |
| FLB_BINARY | Build executable | Yes |
| FLB_CONFIG_YAML | Enable YAML configuration support | Yes |
| FLB_CUSTOM_CALYPTIA | Enable Calyptia Support | Yes |
| FLB_EXAMPLES | Build examples | Yes |
| FLB_HTTP_SERVER | Enable HTTP Server | No |
| FLB_INOTIFY | Enable Inotify support | Yes |
| FLB_JEMALLOC | Use Jemalloc as default memory allocator | No |
| FLB_LUAJIT | Enable Lua scripting support | Yes |
| FLB_METRICS | Enable metrics support | Yes |
| FLB_MTRACE | Enable mtrace support | No |
| FLB_PARSER | Build with Parser support | Yes |
| FLB_POSIX_TLS | Force POSIX thread storage | No |
| FLB_PROXY_GO | Enable Go plugins support | Yes |
| FLB_RECORD_ACCESSOR | Enable record accessor | Yes |
| FLB_REGEX | Build with Regex support | Yes |
| FLB_RELEASE | Build with release mode (-O2 -g -DNDEBUG) | No |
| FLB_RUN_LDCONFIG | Enable execution of ldconfig after installation | No |
| FLB_SHARED_LIB | Build shared library | Yes |
| FLB_SIGNV4 | Enable AWS Signv4 support | Yes |
| FLB_SQLDB | Enable SQL embedded database support | No |

> | FLB_STATIC_CONF | Build binary using static configuration files. The value of this option must be a directory containing configuration files. |  |

Reviewer: The other lines have a value for the last cell in the line, but this one is missing a Yes or No.

Author: Because there isn't ...
https://github.com/fluent/fluent-bit/blob/4db79b40aa930a29298771e8e4ca4e97d9e9fbba/CMakeLists.txt#L139

| FLB_WASM | Build with WASM runtime support | Yes |
| FLB_WASM_STACK_PROTECT | Build with WASM runtime with strong stack protector flags | No |
| FLB_WAMRC | Build with WASM AOT compiler executable | No |
| FLB_WINDOWS_DEFAULTS | Build with predefined Windows settings | Yes |

### Development Options

| option | description | default |
| :--- | :--- | :--- |
| FLB_BACKTRACE | Enable backtrace/stacktrace support | Yes |
| FLB_CHUNK_TRACE | Enable chunk traces | Yes |
| FLB_CORO_STACK_SIZE | Set coroutine stack size | |
| FLB_COVERAGE | Build with code-coverage | No |
| FLB_DEBUG | Build binaries with debug symbols | No |
| FLB_HTTP_CLIENT_DEBUG | Enable HTTP Client debug callbacks | No |
| FLB_SMALL | Minimise binary size | No |
| FLB_TESTS | Enable tests | No |
| FLB_TESTS_RUNTIME | Enable runtime tests | No |
| FLB_TESTS_INTERNAL | Enable internal tests | No |
| FLB_TESTS_INTERNAL_FUZZ | Enable internal fuzz tests | No |
| FLB_TESTS_OSSFUZZ | Enable OSS-Fuzz build | No |
| FLB_TRACE | Enable trace mode | No |
| FLB_VALGRIND | Enable Valgrind support | No |

### Optimization Options

| option | description | default |
| :--- | :--- | :--- |

> | FLB_MSGPACK_TO_JSON_INIT_BUFFER_SIZE | Determine initial buffer size for msgpack to json conversion in terms of memory used by payload. | 2.0 |
> | FLB_MSGPACK_TO_JSON_REALLOC_BUFFER_SIZE | Determine percentage of reallocation size when msgpack to json conversion buffer runs out of memory. | 0.1 |

Reviewer: ```suggestion
| FLB_MSGPACK_TO_JSON_INIT_BUFFER_SIZE | Determine initial buffer size for msgpack to JSON conversion in terms of memory used by payload. | 2.0 |
| FLB_MSGPACK_TO_JSON_REALLOC_BUFFER_SIZE | Determine percentage of reallocation size when msgpack to JSON conversion buffer runs out of memory. | 0.1 |
```
